internal/runners/initialize/init.go

```diff
@@ -33,19 +32,49 @@ type RunParams struct {
 
 // Initialize stores scope-related dependencies.
 type Initialize struct {
-	out output.Outputer
+	config projectfile.ConfigGetter
+	out    output.Outputer
 }
 
 type primeable interface {
+	primer.Configurer
 	primer.Outputer
 }
 
 // New returns a prepared ptr to Initialize instance.
 func New(prime primeable) *Initialize {
-	return &Initialize{prime.Output()}
+	return &Initialize{prime.Config(), prime.Output()}
 }
 
-func sanitize(params *RunParams) error {
+// inferLanguage tries to infer a reasonable default language from the project currently in use
+// (i.e. `state use show`).
+// Error handling is not necessary because it's an input error to not include a language to
+// `state init`. We're just trying to infer one as a convenience to the user.
+func inferLanguage(config projectfile.ConfigGetter) string {
+	defaultProjectDir := config.GetString(constants.GlobalDefaultPrefname)
+	if defaultProjectDir == "" {
+		return ""
+	}
+	defaultProj, err := project.FromPath(defaultProjectDir)
+	if err != nil {
+		return ""
+	}
+	commitID := defaultProj.CommitUUID()
+	if commitID == "" {
+		return ""
+	}
+	lang, err := model.FetchLanguageForCommit(commitID)
+	if err != nil {
+		return ""
+	}
```

Comment:
To keep this logic centralized maybe we should move this to convenience methods under the globaldefault package? Mainly just thinking out loud.

Reply:
It seems a bit specialized, so I'm not sure it's worth hoisting up.